runtime/src/accounts.rs

```diff
@@ -141,8 +142,15 @@ impl Accounts {
                 error_counters.insufficient_funds += 1;
                 Err(TransactionError::InsufficientFundsForFee)
             } else {
-                accounts[0].lamports -= fee;
-                Ok((accounts, rents))
+                match hash_age_kind {
+                    HashAgeKind::FullSpend if accounts[0].lamports != fee => {
```

Comment:
@mvines @garious 

is the fee constant for the duration of the epoch?  The book implies its tied to a blockhash.  https://docs.solana.com/book/implemented-proposals/transaction-fees#congestion-driven-fees

If that is true, then we need to spend the full account, otherwise there is a denial of service on custodian keys.

Reply:
Fee is constant for a given blockhash only, it is not constant for an epoch.  One way out is to require `HashAgeKind::FullSpend` always pay the maximum transaction fee

Reply:
ironically these are cheaper, since we don't need to store the signature :)

Reply:
minimum transaction fee then!

Reply:
na, that probably doesn't work :)

Reply:
yea, i think we need to use the max.  But, we can't have the max be 1000x, i think the book has it at 10x.  Any reason why not keep the fee at 10x?   Another way to do it is to fully spend it if its greater then the current fee but less than the max.

Reply:
The max signature fee as implemented today is 10x the target signature fee, not 1000x

Reply:
What's the rational behind the maximum fee?  It seems like that would prevent a market from forming on the off chance we experience sufficiently high congestion as to backlog TXs.  I understand that we're aiming for predictable fees, but it seems like that should be a rate of change limit on the minimum rather than establishing a cap.

To the point, using any hard boundary on the fees enables the DoS, right?  So long as it's an "above X" or "between X and Y" we should be OK

Reply:
Fees are tied to a blockhash and known in advance.  The problem here is we don't have that blockhash to derive that fee from.   Fees change over time based on a rate of change limit and do have a min/max cap, see fee_calculator.rs

Reply:
@t-nelson , last thing we want is for the transaction to not execute but execute later.  

Can you document the "unreliable cancel" mechanism:

Client needs to submit an alternative transaction that spends the "fee account" before this one is encoded.